### ISSUE TITLE
Enhancement Request: Unsharp mask filter

### ISSUE THREAD

**Issue by zaryk** [JimBobSquarePants/ImageProcessor#480]
Would you consider adding Unsharp mask filter?  Here is another project that has it, but would prefer it to be added to ImageProcessor for consistency and flow.

http://kaliko.com/image-library/source-code/

Thanks,

**Comment by JimBobSquarePants** [comment]
You mean for the new Core library? I probably won't be adding any new features to the Framework library.

Should be pretty easy to do. It's essentially blurring a copy and applying the following formula.

https://en.wikipedia.org/wiki/Unsharp_masking#Digital_unsharp_masking

**Comment by zaryk** [comment]
Yea, I saw that after I posted to see if I could do something similar with what you already have.  I was looking at the source for both the libraries, and the kaliko library uses gaussianblur.  At first I was thinking to try something similar to Original + ( Original - Blur) * Amount, but didn't really know where the "Amount" would be applied.  Then I figured that since GaussianSharpen already has similar parameters to the kaliko library, I would try that.  I am able to get a similar effect with radius to 2, sigma 1, threshold 1.  Though it looks a bit too sharp, it may work out.  I am trying to write a program that is similar to Gimps plugin - David's Batch Processor https://sourceforge.net/projects/gimpdbpplugin/, and get similar effect.  The plugin doesn't do recursive folder search though, and I have about 7,000 pngs to modify.  I also want my app to compress the images, which it looks like you have Quantization built in.  I will play around with it more, but image manipulation isn't my strength and don't know the ImageProcessor library that well so I may have more questions.   Thanks for your time!
